src/Store.js

```diff
@@ -149,4 +197,27 @@ export default class Store {
       this.findOne({ [identifier]: itemKey[identifier] }).remove(source)));
   }
 
+  _workFetchResult(storeItems, fetchItems, count, source) {
+    // this is not very performant right now. But as we want to release the first
+    // beta and aren't sure if this behavior stays the same anyway we go with this for now.
+    const identifier = this.schema.getKeyIdentifierFor(source.AS_TARGET);
+    const itemsToRemove = [];
+    storeItems.forEach((storeItem) => {
+      if (storeItem[identifier]) {
+        const fetchItem = fetchItems.find(item => storeItem[identifier] === item[identifier]);
```

Comment:
Maybe in the future we can create a temporary map of fetchItems first. Then we don't need to do this every storeItem, also we wouldn't need fetchItems splice because the performance of map should be good enough